DependencyInjection/Configuration.php

```diff
@@ -946,6 +954,17 @@ public function getConfigTreeBuilder()
                             ->ifTrue(function ($v) { return 'predis' === $v['type'] && empty($v['redis']); })
                             ->thenInvalid('The host has to be specified to use a RedisLogHandler')
                         ->end()
+                        ->validate()
```

Comment:
this should be done in `beforeNormalization`, not in `validate` IMO, so that the copy of the options to the new name is done before merging various configs together. This ensures that each config file can migrate independently, instead of loosing all non-migrated config as soon as one file gets migrated.

Reply:
Makes sense, switched to `beforeNormalization`.